libqtile/config.py

```diff
@@ -392,6 +391,22 @@ def _configure(
         self.y = y
         self.width = width
         self.height = height
+
+        if not hasattr(self, "group"):
+            # Ensure that this screen actually *has* a group. It will get this the first
+            # time this method is called, and until then accessing self.group will raise
+            # an AttributeError. The 'right' approach would be typing `self.group` as
+            # `_Group | None` and setting it to `None` upon init, but conceptually no
+            # screens should ever not have a group and this helps us avoid having to
+            # check for `None`s everywhere.
+            for grp in qtile.groups:
+                if grp.screen is None or grp.screen is self:
```

Comment:
Why do we check if the group's `screen` is None? Isn't the `screen` value only set when the group is visible? If that's the case, this check will put the first available group on the screen and ignore the `group` parameter in this method call.

I could have it completely wrong but it just looks odd to me.

Reply:
It is odd, so let's definitely make the comment more clear so that future readers of the code understand the logic, and let's make sure the logic actually does make sense :D 

Basically the idea is that this ensures that when `Qtile._process_screens` runs (config loading, reloading or `cmd_reconfigure_screens`), every screen has a group so that `screen.group` never raises an `AttributeError` (as no group as assigned during `__init__`). _Then_, the desired group (the `group` parameter to this method) is set as the screen's actual group. Thus, any additional operations within `self.set_group(group)` can do their thing. Often, this is a no-op because it'll be the very same screen, but sometimes it won't be, and importantly this is the case that raises the origin `AttributeError` that is sometimes reported, because `self.set_group(group)` tries to access `self.group` before it's every been set to something.

Looking at this again I think it might be better to move this block out of this method and into `Qtile._process_screens` directly to run right before `scr._configure` is called, so it's more like `Qtile` is setting up the screen rather than the screen bootstrapping itself, but maybe that's unimportant.

Reply:
I've moved this block out into `Qtile._process_screens`, reworded the comment a bit, and used the group that is about to get passed into `scr._configure` so hopefully it's more clear what's going on